site/src/layouts/star-begging.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useEffect, useState } from 'react';
+
+import * as styles from './star-begging.module.less';
+
+// Forked from https://github.com/javalin/javalin.github.io/blob/36bc38763043bd9326b1b8c8d7acca84db3e5cc1/_includes/starBegging.html
+
+const StarBegging: React.FC = () => {
+  const [display, setDisplay] = useState(false);
+  useEffect(() => {
+    if (localStorage.getItem('dismissed') !== 'true') {
+      setTimeout(() => setDisplay(true), 10000); // 10 seconds
+    }
+  });
```

Comment:
Otherwise, this will be called every time the component is re-rendered

```suggestion
  }, []);
```